crates/polars-stream/src/nodes/csv_source.rs

```diff
@@ -0,0 +1,675 @@
+                                // lines.
+                                SplitSlicePosition::Before => continue,
+                                SplitSlicePosition::Overlapping(offset, len) => (offset, len),
+                                SplitSlicePosition::After => break 'main,
+                            }
+                        } else {
+                            // (0, 0) is interpreted as no slicing
+                            (0, 0)
+                        };
+
+                        let mut mem_slice_this_chunk =
+                            mem_slice.slice(slice_start..slice_start + position);
+
+                        let morsel_seq = *morsel_seq_ref;
+                        *morsel_seq_ref = morsel_seq.successor();
+
+                        let Some(mut indexed_wait_group) = wait_groups.next().await else {
+                            break;
+                        };
+
+                        let mut path_name = path_name.clone();
+
+                        loop {
+                            use crate::async_primitives::connector::SendError;
+
+                            let channel_index = indexed_wait_group.index();
+                            let wait_token = indexed_wait_group.token();
+
+                            match line_batch_senders[channel_index].try_send(LineBatch {
+                                bytes: mem_slice_this_chunk,
+                                n_lines: count,
+                                slice,
+                                row_offset: current_row_offset,
+                                morsel_seq,
+                                wait_token,
+                                path_name,
+                            }) {
+                                Ok(_) => {
+                                    wait_groups.push(indexed_wait_group.wait());
+                                    break;
+                                },
+                                Err(SendError::Closed(v)) => {
+                                    mem_slice_this_chunk = v.bytes;
+                                    path_name = v.path_name;
+                                },
+                                Err(SendError::Full(_)) => unreachable!(),
+                            }
+
+                            let Some(v) = wait_groups.next().await else {
+                                break 'main; // All channels closed
+                            };
+
+                            indexed_wait_group = v;
+                        }
+                    }
+                }
+
+                Ok(())
+            }),
+        );
+
+        (
+            line_batch_receivers,
+            // TODO: Refactor so that we don't unwrap, it's currently hard because
+            // `ComputeNode::{initialize, spawn}` doesn't return a `PolarsResult`
+            Arc::new(self.try_init_chunk_reader().unwrap()),
```

Comment:
This `unwrap()` is not ideal, but I don't currently have any good ideas on how to avoid this